src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -255,20 +255,13 @@ public void setSheetPos(String sheetName, int pos) {
     }
 
     public String[] getDefinedNames(boolean validOnly) {
-        //TODO getNameAt is deprecated in POI 4.0 -
-        // "New projects should avoid accessing named ranges by index."
-        int count = workbook.getNumberOfNames();
-        // String[] nameNames = new String[count];
         ArrayList<String> nameNames = new ArrayList<String>();
-
-        for(int i = 0; i < count; i++) {
-            Name namedRegion = workbook.getNameAt(i);
+        for(Name namedRegion : workbook.getAllNames()) {
             // if valid only, check corresponding reference formula validity
             if(validOnly && !isValidNamedRegion(namedRegion)) continue;
             nameNames.add(namedRegion.getNameName());
         }
-
-        return nameNames.toArray(new String[nameNames.size()]);
+        return nameNames.toArray(new String[0]);
```

Comment:
Oh, wasn't aware of this use of `toArray` - nice

Reply:
Intellij highlighted it - looks like passing String[0] is more performant in this case.